google/cloud/storage/internal/curl_handle.cc

```diff
@@ -220,41 +220,167 @@ Status CurlHandle::AsStatus(CURLcode e, char const* where) {
+
+    case CURLE_PARTIAL_FILE:
+      code = StatusCode::kUnavailable;
+      break;
+
+    case CURLE_FTP_COULDNT_RETR_FILE:
+    case CURLE_QUOTE_ERROR:
+    case CURLE_WRITE_ERROR:
+    case CURLE_UPLOAD_FAILED:
+    case CURLE_READ_ERROR:
+    case CURLE_OUT_OF_MEMORY:
+      code = StatusCode::kUnknown;
+      break;
+
     case CURLE_OPERATION_TIMEDOUT:
       code = StatusCode::kDeadlineExceeded;
       break;
+
+    case CURLE_FTP_PORT_FAILED:
+    case CURLE_FTP_COULDNT_USE_REST:
+      code = StatusCode::kUnknown;
+      break;
+
     case CURLE_RANGE_ERROR:
       // This is defined as "the server does not *support or *accept* range
       // requests", so it means something stronger than "your range value is
       // not valid".
       code = StatusCode::kUnimplemented;
       break;
+
+    case CURLE_HTTP_POST_ERROR:
+      code = StatusCode::kUnknown;
+      break;
+
+    case CURLE_SSL_CONNECT_ERROR:
+      code = StatusCode::kUnavailable;
+      break;
+
     case CURLE_BAD_DOWNLOAD_RESUME:
       code = StatusCode::kInvalidArgument;
       break;
+
+    case CURLE_FILE_COULDNT_READ_FILE:
+    case CURLE_LDAP_CANNOT_BIND:
+    case CURLE_LDAP_SEARCH_FAILED:
+    case CURLE_FUNCTION_NOT_FOUND:
+      code = StatusCode::kUnknown;
+      break;
+
     case CURLE_ABORTED_BY_CALLBACK:
       code = StatusCode::kAborted;
       break;
+
+    case CURLE_BAD_FUNCTION_ARGUMENT:
+    case CURLE_INTERFACE_FAILED:
+    case CURLE_TOO_MANY_REDIRECTS:
+    case CURLE_UNKNOWN_OPTION:
+    case CURLE_TELNET_OPTION_SYNTAX:
+      code = StatusCode::kUnknown;
+      break;
+
+    case CURLE_GOT_NOTHING:
+      code = StatusCode::kUnavailable;
+      break;
+
+    case CURLE_SSL_ENGINE_NOTFOUND:
+      code = StatusCode::kUnknown;
+      break;
+
+    case CURLE_RECV_ERROR:
+    case CURLE_SEND_ERROR:
+      code = StatusCode::kUnavailable;
+      break;
+
+    case CURLE_SSL_CERTPROBLEM:
+    case CURLE_SSL_CIPHER:
+    case CURLE_PEER_FAILED_VERIFICATION:
+    case CURLE_BAD_CONTENT_ENCODING:
+    case CURLE_LDAP_INVALID_URL:
+    case CURLE_FILESIZE_EXCEEDED:
+    case CURLE_USE_SSL_FAILED:
+    case CURLE_SEND_FAIL_REWIND:
+    case CURLE_SSL_ENGINE_SETFAILED:
+    case CURLE_LOGIN_DENIED:
+    case CURLE_TFTP_NOTFOUND:
+    case CURLE_TFTP_PERM:
+    case CURLE_REMOTE_DISK_FULL:
+    case CURLE_TFTP_ILLEGAL:
+    case CURLE_TFTP_UNKNOWNID:
+    case CURLE_REMOTE_FILE_EXISTS:
+    case CURLE_TFTP_NOSUCHUSER:
+    case CURLE_CONV_FAILED:
+    case CURLE_CONV_REQD:
+    case CURLE_SSL_CACERT_BADFILE:
+      code = StatusCode::kUnknown;
+      break;
+
     case CURLE_REMOTE_FILE_NOT_FOUND:
       code = StatusCode::kNotFound;
       break;
+
+    case CURLE_SSH:
+    case CURLE_SSL_SHUTDOWN_FAILED:
+    case CURLE_AGAIN:
+      // Only returned by curl_easy_{recv,send}, but should not with the
+      // configuration we use for libcurl. And the recovery action is to call
+      // curl_easy_{recv,send} again, but it is too late to do that here.
```

Comment:
This comment is difficult to associate with the relevant cases cases.  Maybe the case should just be split, like some others?

Reply:
I agree, thank you, fixed.